packages/webview_flutter/lib/webview_flutter.dart

```diff
@@ -121,6 +125,28 @@ class WebView extends StatefulWidget {
   })  : assert(javascriptMode != null),
         super(key: key);
 
+  static WebViewPlatformInterface _implementation;
+
+  static set implementation(WebViewPlatformInterface implementation) {
+    _implementation = implementation;
+  }
+
+  static WebViewPlatformInterface get implementation {
+    if (_implementation != null) {
+      return implementation;
+    }
+
+    if (defaultTargetPlatform == TargetPlatform.android) {
+      return WebViewAndroidImplementation();
+    }
+
+    if (defaultTargetPlatform == TargetPlatform.iOS) {
+      return WebViewIosImplementation();
+    }
+
+    throw UnsupportedError("Trying to use the default webview implementation for $defaultTargetPlatform but there isn't a default one");
+  }
```

Comment:
did you intend to cache the returned value?

Reply:
done